### PR TITLE
Document tileset traversal functions

### PR DESCRIPTION
**Pull request by jjhembd**
This PR cleans up and documents helper functions in `Cesium3DTile`, `Cesium3DTileset`, and `Cesium3DTilesetTraversal`, to make the tileset traversal logic easier to follow.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @jjhembd!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/tileset-traversal/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: packages/engine/Source/Scene/Cesium3DTile.js [hunk 1]
```diff
@@ -52,6 +52,9 @@ import Pass from "../Renderer/Pass.js";
  *
  * @alias Cesium3DTile
  * @constructor
+ * @param {Cesium3DTileset} tileset
+ * @param {Resource} baseResource
```
Comment (ggetz) [inline packages/engine/Source/Scene/Cesium3DTile.js:56]: This is missing the `header` parameter.

**Review by ggetz — COMMENTED** [review]
Thanks @jjhembd. I noticed a unit test failure that will need to be fixed- `tile with tileset content expires`

File: packages/engine/Source/Scene/Cesium3DTileset.js [hunk 2]
```diff
@@ -2694,8 +2753,9 @@ function updateTiles(tileset, frameState, passOptions) {
     statistics.incrementSelectionCounts(tile.content);
     ++statistics.selected;
   }
-  for (i = 0; i < emptyLength; ++i) {
```
Comment (jjhembd) [inline packages/engine/Source/Scene/Cesium3DTileset.js:2697]: See "Declare variables where they are first used" in the [coding guide](https://github.com/CesiumGS/cesium/blob/main/Documentation/Contributors/CodingGuide/README.md#basic-code-construction)

**Comment by ggetz** [comment]
Thanks @jjhembd for the updates for clarity!

**Pull request merged**